languages/ocaml.json

```
@@ -26,5 +26,35 @@
     "decreaseIndentPattern": "^\\s*(end|done|with|in|else)\\b|^\\s*;;",
     "increaseIndentPattern": "\\b(begin|object)\\s*$|\\blet [a-zA-Z0-9_-]+( [^ ]+)* =\\s*$|method!?[ \\t]+.*=[ \\t]*$|->[ \\t]*$|\\b(for|while)(_lwt)?[ \\t]+.*[ \\t]+do[ \\t]*$|(\\btry(_lwt)?$|\\bif\\s+.*\\sthen$|\\belse|[:=]\\s*sig|=\\s*struct)\\s*$",
     "indentNextLinePattern": "(?!\\bif.*then.*(else.*|(;|[ \\t]in)[ \\t]*$))\\bif|\\bthen[ \\t]*$|\\belse[ \\t]*$$"
-  }
+  },
+  "onEnterRules": [
+    {
+      "beforeText": "\\(\\*\\*",
+      "action": {
+        "indent": "none",
+        "appendText": "    "
+      }
+    },
+    {
+      "beforeText": "\\(\\*",
+      "action": {
+        "indent": "none",
+        "appendText": "   "
+      }
+    },
+    {
+      "beforeText": "\\*\\)",
+      "action": {
+        "indent": "outdent",
+        "removeText": 3
```

Did the snippet I provide not work?

This currently removes too many spaces. `outdent` removes up to one indentation level so 1 or 2 spaces. And then `removeText` removes an _additional_ 3 spaces for a total of 4 or 5 spaces removed.

We need to handle 3 or 4 spaces, so the `removeText` needs to be smaller.

```suggestion
        "indent": "outdent",
        "removeText": 2
```

---

I didn't have much time so I tried to do it in one go but it didn't work out.